arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,31 @@
 #include <Smartcar.h>
 
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, 6, 7, 200);
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
 void setup() {
-  // put your setup code here, to run once:
+  Serial.begin(9600);
+  Serial.setTimeout(280);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
+  car.setSpeed(90);
+  preventCrash();
+}
+
+void preventCrash() {
+  unsigned int fDistance = front.getDistance();
+  if (fDistance > 0) {
```

Comment:
This will make the car stop at **any** distance detected. You should:

> make the Smartcar move forward until it detects an obstacle **at a distance of your choosing.**

Reply:
Yes, but this is bounded by the detection range of the SR04 front, which is set to 200. I will make it more explicit though.

Reply:
If you really intended the distance to be `200` it would have indeed worked but readability wise one would not understand your intent. :)
This is normally read as "if any distance is detected".